tests/operator_test.go

```diff
@@ -1434,6 +1435,66 @@ spec:
 			Expect(monitoringLabel).To(Equal("true"))
 		})
 	})
+
+	// This test isn't technically testing anything with virt-operator, but all the framework
+	// is in place to tear down and re-set the environment when it's been modified,
+	// and this test does modify significant components.
+	Context("With selinuxLauncherType defined", func() {
+		It("Should honor custom SELinux type for virt-launcher", func() {
+			cfgMap, err := virtClient.CoreV1().ConfigMaps(tests.KubeVirtInstallNamespace).Get("kubevirt-config", metav1.GetOptions{})
+			// Update KubeVirt's ConfigMap
+			if err != nil && !errors.IsNotFound(err) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			superPrivilegedType := "spc_t"
+
+			By("Starting a VMI")
+			vmi := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("Fetching virt-launcher Pod")
+			pod := tests.GetPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Verifying SELinux context contains default type")
+			Expect(pod.Spec.SecurityContext.SELinuxOptions.Type).ToNot(Equal(superPrivilegedType))
+
+			By("Deleting the VMI")
+			err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.Name, &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Updating ConfigMap")
+			cfgMap.Data[virtconfig.SELinuxLauncherTypeKey] = superPrivilegedType
+
+			newData, err := json.Marshal(cfgMap.Data)
+			Expect(err).ToNot(HaveOccurred())
+			data := fmt.Sprintf(`[{ "op": "replace", "path": "/data", "value": %s }]`, string(newData))
+
+			cfgMap, err = virtClient.CoreV1().ConfigMaps(tests.KubeVirtInstallNamespace).Patch("kubevirt-config", types.JSONPatchType, []byte(data))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Deleting virt-controller deployment")
+			err = virtClient.AppsV1().Deployments(tests.KubeVirtInstallNamespace).Delete("virt-controller", &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Starting a New VMI")
+			vmi = tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("Fetching virt-launcher Pod")
+			pod = tests.GetPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Verifying SELinux context contains custom type")
```

Comment:
@stu-gott we should probably at least also check if we can log in to the machine?